lnbits/core/models/offers.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import annotations
+
+from datetime import datetime, timezone
+
+from pydantic import BaseModel, Field
+
+from lnbits.db import FilterModel
+from lnbits.wallets import get_funding_source
+from lnbits.wallets.base import (
+    OfferStatus,
+)
+
+
+class CreateOffer(BaseModel):
+    memo: str
+    amount_msat: int | None = None
+    extra: dict | None = {}
+    expiry: datetime | None = None
+    webhook: str | None = None
+
+
+class Offer(BaseModel):
+    offer_id: str
+    wallet_id: str
+    amount: int | None = None
+    active: bool
+    single_use: bool
+    used: bool
+    bolt12: str
+    memo: str | None = None
+    expiry: datetime | None = None
+    webhook: str | None = None
+    webhook_status: str | None = None
+    tag: str | None = None
+    extension: str | None = None
+    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
+    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
+    extra: dict = {}
+
+    @property
+    def is_active(self) -> bool:
+        return not self.active == False  # noqa E712
+
+    @property
+    def is_inactive(self) -> bool:
+        return self.active == False  # noqa E712
+
+    @property
+    def is_single_use(self) -> bool:
+        return not self.single_use == False  # noqa E712
+
+    @property
+    def is_multiple_use(self) -> bool:
+        return self.single_use == False  # noqa E712
+
+    @property
+    def is_used(self) -> bool:
+        return self.used == True  # noqa E712
+
+    @property
+    def is_unused(self) -> bool:
+        return not self.used == True  # noqa E712
+
+    @property
+    def msat(self) -> int | None:
+        return self.amount
+
+    @property
+    def sat(self) -> int | None:
+        return int(self.amount / 1000) if self.amount else None
+
+    @property
+    def is_expired(self) -> bool:
+        return self.expiry < datetime.now(timezone.utc) if self.expiry else False
+
+    async def check_status(self) -> OfferStatus:
+        funding_source = get_funding_source()
+        status = await funding_source.get_offer_status(self.offer_id)
+        return status
```

Comment:
can we remove this `check_status`? i know its takes from `payment.get_status` but i really want to get rid of it on payment also.

can we do in `services/payments.py` maybe add a `check_offer_status()` there and pass the offer_id.